gutenberg/acquire/text.py

```diff
@@ -37,15 +41,28 @@ def _etextno_to_uri_subdirectory(etextno):
     return subdir
 
 
-def _format_download_uri(etextno):
+def _format_download_uri(etextno, mirror=None):
     """Returns the download location on the Project Gutenberg servers for a
     given text.
 
     Raises:
         UnknownDownloadUri: If no download location can be found for the text.
-
     """
-    uri_root = r'http://www.gutenberg.lib.md.us'
+    global _MIRROR_CHECK
```

Comment:
Instead of using a global, we could extract the mirror-check out into its own function, e.g. `_check_mirror_exists`, and then separate out the concern to only make the function execute once, e.g. via a decorator like this:

```py
 def execute_only_once(func):
   def call_once(*args, **kwargs):
     if not call_once._called:
       try:
         return func(*args, **kwargs)
       finally:
         call_once._called = True
   call_once._called = False
   return call_once

@execute_only_once
def _check_mirror_exists():
  # implementation here
```

[Code source](http://pydev.blogspot.ca/2012/12/python-tricks-making-sure-function-is.html)